lib/datasource/index.ts

```diff
@@ -201,7 +201,7 @@ async function fetchReleases(
     }
     logError(datasource.id, config.lookupName, err);
   }
-  if (!dep || _.isEqual(dep, { releases: [] })) {
+  if (!dep || equal(dep, { releases: [] })) {
```

Comment:
@rarkins Do we really need a deep equal here? Can't we simply check for `!dep?.releases?.length` ?

Reply:
No, there's a case for Go where it has empty releases but a valid sourceUrl. We later generate digests. So it's intentionally checking that only a truly empty object gets removed.